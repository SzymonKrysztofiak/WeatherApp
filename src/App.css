@import url("https://fonts.googleapis.com/css?family=Open+Sans:400,700&subset=latin-ext");

body {
    margin: 0;
    padding: 0;
    border: 0;
    box-sizing: border-box;
    font-family: "Open Sans", sans-serif;
    color: rgb(255, 255, 255);
}

.bg-container {
    background: url("./img/background.jpg") center center no-repeat;
    position: fixed;
    top: 50%;
    left: 50%;
    transform: translateX(-50%) translateY(-50%);
    background-size: cover;
    width: 100%;
    height: 100vh;
    z-index: -100;
}

.flex-container {
    display: flex;
    align-items: center;
    flex-direction: column;
    padding: 10px;
    height: auto;
}

/* HEADER */

.title-container {
    max-width: 600px;
    margin-top: 20vh;
}

.title-container__title {
    font-size: 72px;
    margin-left: 70px;
    margin-bottom: 0;
    text-align: center;
    transition: 0.5s;
}

.title-container__title::before {
    content: "";
    background: url("./img/cloud.png") center center no-repeat;
    background-size: contain;
    height: 40px;
    width: 57px;
    position: absolute;
    margin-top: 22px;
    margin-left: -70px;
    transition: 0.5s;
}

.title-container__description {
    margin-top: 25px;
    line-height: 34px;
    font-size: 20px;
    text-align: center;
}

.title-container__description > span {
    font-weight: 700;
}

/* FORM */

.form {
    max-width: 600px;
    width: 100%;
    margin-top: 25px;
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    flex-wrap: wrap;
}

.form-input {
    padding: 0;
    padding-left: 20px;
    height: 56px;
    width: 376px;
    background-color: rgba(255, 255, 255, 0.1);
    border: rgb(255, 255, 255) 2px solid;
    color: rgb(255, 255, 255);
    font-size: 18px;
}

.button {
    background-color: rgba(255, 255, 255, 0.2);
    border: none;
    color: rgb(255, 255, 255);
    text-transform: uppercase;
    font-weight: 700;
    font-size: 20px;
    padding: 0;
    cursor: pointer;
}

.form-button {
    height: 60px;
    width: 180px;
}

::-webkit-input-placeholder {
    color: rgba(255, 255, 255, 0.7);
}
::-moz-placeholder {
    color: rgba(255, 255, 255, 0.7);
}
:-ms-input-placeholder {
    color: rgba(255, 255, 255, 0.7);
}
:-moz-placeholder {
    color: rgba(255, 255, 255, 0.7);
}

/* RESULTS */

.results-container {
    display: flex;
    align-items: center;
    margin-top: 40px;
    flex-wrap: wrap;
    margin-bottom: 30px;
}

.results-container__temp > p {
    font-size: 72px;
    font-weight: 700;
    margin: 0 20px;
}

.results-container__rest > p,
.result-container__error > p {
    font-size: 20px;
    margin: 0 20px;
    line-height: 34px;
}

.result-container__error > p {
    max-width: 500px;
    width: 100%;
    margin: 0;
    text-align: center;
}

.result-container__button {
    text-transform: none;
    font-weight: 300;
    height: 30px;
    width: 100%;
    margin: 5px 0 0 0;
    display: block;
}

/* FOOTER */

.footer-description {
    font-size: 14px;
    text-align: center;
}

.footer-description a {
    color: rgb(255, 255, 255);
}

/* LOADER */

.loader {
    background: url("./img/cog.png");
    width: 39px;
    height: 40px;
    animation: spin 2s linear infinite;
}

@keyframes spin {
    0% {
        transform: rotate(0deg);
    }
    100% {
        transform: rotate(360deg);
    }
}

/* MEDIA */

@media (max-width: 600px) {
    .title-container {
        margin-top: 10vh;
    }
    .form-button {
        height: 60px;
        width: 100%;
        margin-top: 25px;
    }
    .form-input {
        width: 100%;
    }
}

@media (max-width: 535px) {
    .title-container__title {
        font-size: 60px;
        margin-left: 60px;
    }
    .title-container__title::before {
        width: 49px;
        height: 35px;
        margin-left: -60px;
        margin-top: 16px;
    }
}

@media (max-width: 492px) {
    .title-container {
        margin-top: 5vh;
    }
    .results-container {
        justify-content: center;
    }
    .results-container__rest > p,
    .title-container__description {
        font-size: 16px !important;
    }
    .results-container__rest {
        margin-top: 25px;
    }
    .title-container__title {
        font-size: 48px;
        margin-left: 50px;
    }
    .title-container__title::before {
        width: 39px;
        height: 28px;
        margin-left: -50px;
        margin-top: 12px;
    }
}

@media (max-width: 370px) {
    .title-container__title {
        font-size: 36px;
        margin-left: 36px;
    }
    .title-container__title::before {
        width: 29px;
        height: 21px;
        margin-left: -36px;
        margin-top: 8px;
    }
}
