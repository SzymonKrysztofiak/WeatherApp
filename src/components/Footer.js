import React from "react";

const Footer = () => {
    return (
        <p className="footer-description">
            Visit my <a href="https://github.com/SzymonKrysztofiak">GitHub</a> /
            Photo by <a href="https://unsplash.com/@lauravinck">Laura Vinck</a>{" "}
            on <a href="https://unsplash.com/">Unsplash</a>
        </p>
    );
};

export default Footer;
