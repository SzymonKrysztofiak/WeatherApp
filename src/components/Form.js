import React from "react";
import api from "../api";

class Form extends React.Component {
    cityRef = React.createRef();

    getGeoCoords = () => {
        if (!navigator.geolocation) {
            const error = "Your browser doesn't support geolocation :(";
            this.props.addError(error);
        } else {
            navigator.geolocation.getCurrentPosition(
                this.geoSuccess,
                this.geoError
            );
        }
    };

    geoSuccess = position => {
        const coords = {
            lat: position.coords.latitude,
            lng: position.coords.longitude
        };
        this.props.addCoordsToState(coords);
    };

    geoError = () => {
        const error = `For some reason we can't read your location :( Check your geolocation settings, or network access.`;
        this.props.addError(error);
    };

    getCoords = async event => {
        event.preventDefault();
        this.props.toggleLoader();
        if (this.cityRef.current.value === "") {
            this.getGeoCoords();
        } else {
            const api_call = await fetch(
                `https://maps.googleapis.com/maps/api/geocode/json?address=${
                    this.cityRef.current.value
                }&key=${api.goog_key}`
            );
            const data = await api_call.json();
            if (data.status === "ZERO_RESULTS") {
                const error = "We don't have any data for this name :(";
                this.props.addError(error);
            } else {
                const coords = await {
                    lat: data.results[0].geometry.location.lat,
                    lng: data.results[0].geometry.location.lng
                };
                this.props.addCoordsToState(coords);
            }
        }
    };

    render() {
        return (
            <form className="form" onSubmit={this.getCoords}>
                <input
                    className="form-input"
                    type="text"
                    name="city"
                    ref={this.cityRef}
                    placeholder="London, Paris, Poznan..."
                />
                <button className="button form-button">Check</button>
            </form>
        );
    }
}

export default Form;
