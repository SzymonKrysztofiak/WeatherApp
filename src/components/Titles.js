import React from "react";

const Titles = () => (
    <header className="title-container">
        <h1 className="title-container__title">
            <span className="glyphicon glyphicon-cloud" aria-hidden="true" />{" "}
            WeatherApp
        </h1>
        <p className="title-container__description">
            Check the weather! Just enter the city name, or leave{" "}
            <span>empty for geolocation</span>...
        </p>
    </header>
);

export default Titles;
