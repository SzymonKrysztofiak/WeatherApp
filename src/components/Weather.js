import React from "react";
import Loader from "./Loader";

const Weather = props => (
    <div className="results-container">
        {props.isLoaded ? null : <Loader />}
        <div className="results-container__temp">
            {props.temperature && <p>{props.temperature}&deg;C</p>}
        </div>
        <div className="results-container__rest">
            {props.date && (
                <p>
                    <strong>Date: </strong>
                    {props.date}
                </p>
            )}
            {props.city &&
                props.country && (
                    <p>
                        <strong>Location: </strong>
                        {props.city}, {props.country}
                    </p>
                )}
            {props.description && (
                <p>
                    <strong>Description: </strong>
                    {props.description}
                </p>
            )}
        </div>
        <div className="result-container__error">
            {props.errorMessage && (
                <p>
                    <strong>{props.errorMessage}</strong>
                </p>
            )}
        </div>
    </div>
);

export default Weather;
