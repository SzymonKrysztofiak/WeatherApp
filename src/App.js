import React from "react";

import Titles from "./components/Titles";
import Form from "./components/Form";
import Weather from "./components/Weather";

const API_KEY = "dbcd93b4e2fe2814afa13caac19f0cc5";

class App extends React.Component {

    state = {
        icon: undefined,
        temperature: undefined,
        city: undefined,
        country: undefined,
        description: undefined,
        error: undefined
    }

    getWeather = async (e) => {
        e.preventDefault();
        const city = e.target.elements.city.value;
        const country = e.target.elements.country.value;
        const api_call = await fetch(`http://api.openweathermap.org/data/2.5/weather?q=${city},${country}&appid=${API_KEY}&units=metric`);
        const data = await api_call.json();
        if(city && country) {
            console.log(data);
            this.setState({
                icon: data.weather[0].icon,
                temperature: data.main.temp,
                city: data.name,
                country: data.sys.country,
                description: data.weather[0].description,
                error: ""
            });
        } else {
            this.setState({
                icon: undefined,
                temperature: undefined,
                city: undefined,
                country: undefined,
                description: undefined,
                error: "Please enter the values."
            });
        }
        console.log(this.getWeather);
    }
    render() {
        return (
            <div className="container">
                <div className="col-xs-12 col-sm-8 col-sm-offset-2 col-md-6 col-md-offset-3">
                    <div className="app-wrapper">
                        <Titles />
                            <main className="main">
                            <section className="form-container">
                                <Form getWeather={this.getWeather} />
                                <Weather 
                                    icon={this.state.icon}
                                    temperature={this.state.temperature} 
                                    city={this.state.city} 
                                    country={this.state.country} 
                                    description={this.state.description} 
                                    error={this.state.error} 
                                />
                            </section>
                        </main>
                    </div>
                </div>
            </div>
        );
    }
};

export default App;