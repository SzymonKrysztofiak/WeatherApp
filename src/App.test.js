import ReactDOM from "react-dom";
import App from "./App";
import Titles from "./components/Titles";
import Form from "./components/Form";
import Weather from "./components/Weather";
import Footer from "./components/Footer";

it("renders without crashing", () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
});

it("includes Titles", () => {
    const wrapper = shallow(<App />);
    expect(wrapper.containsMatchingElement(<Titles />)).toEqual(true);
});

it("includes Form", () => {
    const wrapper = shallow(<App />);
    expect(wrapper.containsMatchingElement(<Form />)).toEqual(true);
});

it("includes Weather", () => {
    const wrapper = shallow(<App />);
    expect(wrapper.containsMatchingElement(<Weather />)).toEqual(true);
});

it("includes Footer", () => {
    const wrapper = shallow(<App />);
    expect(wrapper.containsMatchingElement(<Footer />)).toEqual(true);
});
