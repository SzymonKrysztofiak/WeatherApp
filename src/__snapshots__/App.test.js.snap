// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<React.Fragment>
  <div
    className="bg-container"
  />
  <div
    className="flex-container"
  >
    <Titles />
    <Form
      addCoordsToState={[Function]}
      addError={[Function]}
      toggleLoader={[Function]}
    />
    <Weather
      forecast={Array []}
      isLoaded={true}
    />
    <Footer />
  </div>
</React.Fragment>
`;
